modules/programs/git.nix

```
@@ -330,64 +330,16 @@ in
           '';
         };
 
-        background = mkOption {
-          type = types.enum [
-            "light"
-            "dark"
-          ];
-          default = "light";
-          example = "dark";
-          description = ''
-            Determines whether difftastic should use the lighter or darker colors
-            for syntax highlighting.
-          '';
-        };
-
-        color = mkOption {
-          type = types.enum [
-            "always"
-            "auto"
-            "never"
-          ];
-          default = "auto";
-          example = "always";
-          description = ''
-            Determines when difftastic should color its output.
-          '';
-        };
-
-        context = mkOption {
-          type = types.ints.u32;
-          default = 3;
-          example = 5;
-          description = ''
-            Determines the number of contextual lines to show around changed lines.
-          '';
-        };
-
-        display = mkOption {
-          type = types.enum [
-            "side-by-side"
-            "side-by-side-show-both"
-            "inline"
-          ];
-          default = "side-by-side";
-          example = "inline";
-          description = ''
-            Determines how the output displays - in one column or two columns.
-          '';
-        };
-
-        extraArgs = mkOption {
-          type = types.nullOr (types.listOf types.str);
-          default = null;
-          example = [
-            "--tab-width=8"
-            "--sort-paths"
-          ];
-          description = ''
-            Extra command line arguments to pass to {command}`difft`.
-          '';
+        options = mkOption {
+          type =
+            with lib.types;
+            attrsOf (oneOf [
+              str
+              number
+              bool
+            ]);
+          default = { };
```

```suggestion
          default = { };
          example = {
            background = "dark";
            color = "always";
            context = 5;
            display = "inline";
            tab-width = 8;
            sort-paths = true;
          };
```

Something like this, I was thinking. Just to show some of the "common" / deprecated options new usages.